graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/node/GqlJcrNodeImpl.java

```diff
@@ -562,6 +572,24 @@ public String getThumbnailUrl(@GraphQLName("name") @GraphQLDescription("Thumbnai
         return node.getThumbnailUrl(name);
     }
 
+    @Override
+    @GraphQLDescription("Get render URL")
+    public String getRenderUrl(@GraphQLName("workspace") @GraphQLDescription("The target workspace") @GraphQLNonNull NodeQueryExtensions.Workspace workspace,
+                               @GraphQLName("language") @GraphQLDescription("The language content is rendered in") @GraphQLNonNull String language,
+                               @GraphQLDefaultValue(GqlUtils.SupplierFalse.class) @GraphQLName("findDisplayable") @GraphQLDescription("Finds displayable node") Boolean findDisplayable,
+                               DataFetchingEnvironment environment) {
+        try {
+            String url =  getNodeURL(this.node, workspace.getValue(), LanguageCodeConverters.languageCodeToLocale(language), findDisplayable);
+            HttpServletResponse httpServletResponse = ContextUtil.getHttpServletResponse(environment.getGraphQlContext());
+            HttpServletRequest httpServletRequest = ContextUtil.getHttpServletRequest(environment.getGraphQlContext());
+            UrlRewriteService urs = (UrlRewriteService) SpringContextSingleton.getBean("UrlRewriteService");
+            HttpServletResponse resp = new UrlRewriteWrappedResponse(httpServletResponse, httpServletRequest, urs.getEngine());
+            return resp.encodeURL(url).replace("/modules", "");
```

Comment:
this /modules replace looks strange. Not sure to understand why it's necessary, consider adding comments in code to explain this replace.

Reply:
Thanks for the comment, still I have the feeling that something is wrong about this replace, please take a look at: https://github.com/Jahia/graphql-core/pull/443#discussion_r1691031631